shared/lib/sharing/publicize/selectors.js

```diff
@@ -18,6 +18,23 @@ export function getConnectionsBySiteId( state, siteId ) {
 }
 
 /**
+ * Returns an array of known connections for the given site ID
+ * that are available to the current user
+ *
+ * @param  {Object} state         Global state tree
+ * @param  {Number} siteId        Site ID
+ * @param  {Number} currentUserID ID of the current user
+ * @return {Array}                Site connections
+ */
+export function getConnectionsBySiteIdAvailableToCurrentUser( state, siteId, currentUserID ) {
```

Comment:
Looks like we have a [test suite](https://github.com/Automattic/wp-calypso/blob/fix/editor-sharing-do-not-show-non-shared-connections/shared/lib/sharing/publicize/test/selectors.js) for this file - could we add some tests for this new selector?